src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -547,6 +551,26 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         }
     }
 
+    private fun checkSuperTraitImplemented(holder: RsAnnotationHolder, impl: RsImplItem, trait: RsTraitItem) {
+        if (impl.project.macroExpansionManager.macroExpansionMode !is MacroExpansionMode.New) return
+
+        val typeRef = impl.typeReference ?: return
+        val traitRef = impl.traitRef ?: return
+        val type = typeRef.type
+        val supertraits = trait.typeParamBounds?.polyboundList?.mapNotNull { it.bound } ?: return
+        val lookup = typeRef.implLookup
+
+        for (bound in supertraits) {
+            val boundTrait = bound.traitRef?.resolveToBoundTrait() ?: continue
+            val localBound = boundTrait.substitute(impl.implementedTrait?.subst ?: emptySubstitution)
```

Comment:
Let's extract `impl.implementedTrait?.subst` out of the loop. Also it can be replaced with `traitRef.resolveToBoundTrait()`

Reply:
Well, let's also suppress this check if implemented trait substitution contains unresolved types:
![image](https://user-images.githubusercontent.com/3221931/84056787-62396e00-a9bf-11ea-81f9-ba28b1fff574.png)